app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/LibraryFactory.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.nav.destination.library
+
+import org.kiwix.kiwixlib.Library
+import javax.inject.Inject
+
+class LibraryFactory @Inject constructor() : Library() {
```

Comment:
This is now probably overkill, before we were trying to express that it was important for the functioning of the server that new libraries are created and if we go the KiwixServer.Factory route then that'll express that for us

Reply:
It has been added as a compensation to the deletion of `providesLibrary()` fun from serviceModule. I can remove it from KiwixServer, but is its entire deletion necessary?

Reply:
yes, if you create a KiwixServer.Factory then you would have to pass in this LibraryFactory to it for it to be used and then I will have read the word Factory too much and will hate the code. It is also currently inconsistent, you use a factory for the library but not for the JniKiwixServer and if you were to add a JniKiwixServerFactory that would be altogether too many factories.

Reply:
Fine, so I'll remove LibFactory then